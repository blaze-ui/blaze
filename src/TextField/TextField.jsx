import React, { Component } from "react";
import PropTypes from "prop-types";

import "./inputStyles.scss";
export default class TextField extends Component {
  render() {
    return (
      <div>
        <label> {this.props.children}</label>
        <input className="input" />
      </div>
    );
  }
}
TextField.propTypes = {
  label: PropTypes.string
};
