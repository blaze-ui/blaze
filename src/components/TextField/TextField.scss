@import "../../styles/ColorSchema.scss";

.input {
  width: 95%;
  height: 35px;
  padding-left: 15px;
  border: none;
  border-radius: 5px;
  margin-bottom: 20px;
  background: grey;
  font-family: $zapFont;
}
label {
  font-weight: 100;
  text-align: left;
  margin-left: 13px;
  font-family: $zapFont;
  text-transform: uppercase;
  font-size: 13px;
  padding-left: 15px;
  padding-bottom: 2px;
  color: 	#00D9FF;
  display: block;
}
