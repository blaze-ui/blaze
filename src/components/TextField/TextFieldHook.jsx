import React, {useState} from 'react';
import "./inputStyles.scss";

function TextFieldHook(){
         const [input, setInput] = useState("");
    
        const handleChange = (e) => {
          e.preventDefault();
          setInput(e.target.value);
        }
    return(
        <div>
            <label>Input</label>
            <input 
              className="input"
              type="text" 
              name="text" 
              onChange={handleChange} 
              value={input} 
              placeholder="hook input" />
        </div>
    )
}

export default TextFieldHook;