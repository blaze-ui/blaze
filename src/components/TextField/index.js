/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable linebreak-style */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/button-has-type */
/* eslint-disable react/prefer-stateless-function */
import React from "react";
import PropTypes from "prop-types";
import "./TextField.css";

function TextField(props) {
  return (
    <div>
      <label>{props.children}</label>
      <input className="input" />
    </div>
  );
}
TextField.propTypes = {
  label: PropTypes.string
};

export default TextField;
