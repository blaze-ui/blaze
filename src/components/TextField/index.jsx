import React, { Component } from "react";
import PropTypes from "prop-types";

import "./TextField.css";
function TextField(props) {
const {children }= props;
    return (
      <div>
        <label> {props.children}</label>
        <input className="input" />
      </div>
    );
  
}
TextField.propTypes = {
  label: PropTypes.string
};

export default TextField;