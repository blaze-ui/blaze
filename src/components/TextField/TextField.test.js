import React from "react";
import renderer from "react-test-renderer";
import TextField from "./index";

test("Container tests", () => {
    const component = renderer.create(
        <TextField
            backgroundColor="teal"
            color="white"
            placeholderText="Color Customized"
        >
            Input Hook with custom placeholder
        </TextField>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
