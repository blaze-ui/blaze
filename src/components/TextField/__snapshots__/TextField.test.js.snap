// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container tests 1`] = `
<div>
  <input
    className="input"
    name="text"
    onChange={[Function]}
    placeholder="Color Customized"
    style={
      Object {
        "backgroundColor": "teal",
        "color": "white",
      }
    }
    type="text"
    value=""
  />
   
  <label
    style={
      Object {
        "color": "white",
      }
    }
  >
    Input Hook with custom placeholder
  </label>
</div>
`;
